sdk/log/provider.go

```diff
@@ -14,13 +18,79 @@ import (
 // Compile-time check LoggerProvider implements log.LoggerProvider.
 var _ log.LoggerProvider = (*LoggerProvider)(nil)
 
+const (
+	defaultAttributeCountLimit       = 128
+	defaultAttributeValueLengthLimit = -1
+
+	envarAttributeCountLimit       = "OTEL_LOGRECORD_ATTRIBUTE_COUNT_LIMIT"
+	envarAttributeValueLengthLimit = "OTEL_LOGRECORD_ATTRIBUTE_VALUE_LENGTH_LIMIT"
+)
+
+type limit struct {
+	value int
+	set   bool
+}
+
+func newLimit(value int) limit {
+	return limit{value: value, set: true}
+}
+
+// resolve returns the limit value. If the limit value is set it will return
+// that value. If not it will use the value of the envar if set. Otherwise,
+// fallback is returned.
+func (l limit) resolve(envar string, fallback int) limit {
+	if l.set {
+		return l
+	}
+
+	if v := os.Getenv(envar); v != "" {
+		if n, err := strconv.Atoi(v); err != nil {
+			otel.Handle(fmt.Errorf("invalid %s value %v: %w", envar, v, err))
+		} else {
+			return newLimit(n)
+		}
+	}
+
+	return newLimit(fallback)
+}
+
+type providerConfig struct {
+	resource         *resource.Resource
+	processors       []Processor
+	countLimit       limit
+	valueLengthLimit limit
+}
+
+func newProviderConfig(opts []LoggerProviderOption) providerConfig {
+	var c providerConfig
+	for _, opt := range opts {
+		c = opt.apply(c)
+	}
+
+	if c.resource == nil {
+		c.resource = resource.Default()
+	}
+
+	c.countLimit = c.countLimit.resolve(
+		envarAttributeCountLimit,
+		defaultAttributeCountLimit,
+	)
```

Comment:
Shouldn't the env var resolution be done before applying configs? Otherwise the env vars will override what is configured using options.